portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -77,13 +79,16 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     // collects comment data from the form
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-      Comment comment = makeComment(request);
-      
+      String userId = userService.getCurrentUser().getUserId();
+      Comment comment = makeComment(request); // this comment is basically to get help extract the
+                                              // values more cleanly to make an entity
       
       if (comment != null) {
         Entity commentEntity = new Entity("Comment");
         commentEntity.setProperty("content", comment.getContent());
         commentEntity.setProperty("timestamp", comment.getTimestamp());
+        commentEntity.setProperty("email", comment.getEmail());
+        commentEntity.setProperty("userId", userId);
```

Comment:
couldn't you define and use comment.getUserId()?

Reply:
done!